File: ms-notificacoes/src/main/java/br/edu/ifsp/carona/notificacoes/entity/Notificacao.java
package br.edu.ifsp.carona.notificacoes.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "notificacoes")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notificacao {

    public enum Tipo {
        USUARIO_CADASTRADO,
        VIAGEM_CRIADA,
        PASSAGEIRO_ENTROU
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long destinatarioId;

    @Enumerated(EnumType.STRING)
    private Tipo tipo;

    @Column(nullable = false)
    private String titulo;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String mensagem;

    @Builder.Default
    private Boolean lida = false;

    private String dadosExtras;

    @CreationTimestamp
    private LocalDateTime criadoEm;
}

File: mobile/lib/models/notificacao.dart
class Notificacao {
  final int? id;
  final int? destinatarioId;
  final String tipo;
  final String titulo;
  final String mensagem;
  final bool lida;
  final String? criadoEm;

  Notificacao({
    this.id,
    this.destinatarioId,
    required this.tipo,
    required this.titulo,
    required this.mensagem,
    required this.lida,
    this.criadoEm,
  });

  factory Notificacao.fromJson(Map<String, dynamic> json) {
    return Notificacao(
      id: json['id'] as int?,
      destinatarioId: json['destinatarioId'] as int?,
      tipo: json['tipo'] as String? ?? 'INFO',
      titulo: json['titulo'] as String? ?? '',
      mensagem: json['mensagem'] as String? ?? '',
      lida: json['lida'] as bool? ?? false,
      criadoEm: json['criadoEm'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'destinatarioId': destinatarioId,
      'tipo': tipo,
      'titulo': titulo,
      'mensagem': mensagem,
      'lida': lida,
      'criadoEm': criadoEm,
    };
  }
}
